tools/aws_lab_setup/roles/reachability/tasks/main.yml

```diff
@@ -0,0 +1,4 @@
+- name: Wait 400 seconds, but only start checking after 30 seconds
```

Comment:
@tima can you elaborate on what an "anti-pattern" is?  From reading it seems like it is defined as "is a common response to a recurring problem that is usually ineffective and risks being highly counterproductive."  I am curious how this role is counter-productive, or if you think it will just cause confusion?  The point of roles is to have very simple playbooks that do one thing well.  The reason I made it into a role is it is not clear where this should fit.

The manage_ec2_instances operates on AWS.  It seems like it would be a bad idea to run on AWS, then switch mid-role to connect to the control nodes to see if they are working.  Obviously an include_playbook could work here but that looks even more awkward to me.

The next role we need is user_accounts and common (pretty much unmodified by me).  I could move the wait timer to the beginning of user_accounts and common but they are two separate roles.  I also thought we would re-use the waiting in other places so I just made it a role.  

To be fair I thought this was slightly awkward.... but the AWS api does not seem to have a clean way for checking if the control nodes are actually up and working.  This wait_for_connection seems to be working really well, I just don't know the best place to put it.  From the provision_lab.yml it now looks really clean at a high level

- create the workshops on AWS
- don't do anything until all the ansible(control) nodes are responding to ansible (wait for connection vs wait_for)
- run common role
- run user role
- etc

let me know which direction you want to go...